server/src/main/java/com/cloud/storage/snapshot/SnapshotManagerImpl.java

```diff
@@ -553,6 +553,25 @@ private void postCreateRecurringSnapshotForPolicy(long userId, long volumeId, lo
         }
     }
 
+    public void markFailedSnapshot(long snapshotId) {
+        Account caller = CallContext.current().getCallingAccount();
+
+        // Verify parameters
+        SnapshotVO snapshotCheck = _snapshotDao.findById(snapshotId);
+
+        if (snapshotCheck == null) {
+            throw new InvalidParameterValueException("unable to find a snapshot with id " + snapshotId);
+        }
+
+        _accountMgr.checkAccess(caller, null, true, snapshotCheck);
+
+        snapshotCheck.setState(Snapshot.State.Error);
```

Comment:
@Spaceman1984 can you handle this in the copy callback, and pass the proper event to set the snapshot state?

Reply:
Not sure what you mean @sureshanaparti

Reply:
> Not sure what you mean @sureshanaparti

@Spaceman1984 the caller of _copyAsync_ method would set a callback method to process the result and set the object state accordingly. Can you use that callback method (of snapshot copy operation) to set the state, through the event (check the snapshot state machine) ?

Reply:
I'll take a look.

Reply:
I have removed setting the error state, this leaves the snapshot in a "backing up" state forever which is fine.